web/src/app/components/header/header.component.ts

```diff
@@ -45,23 +45,31 @@ export class HeaderComponent {
   isPublishingChanges = false;
   canManage = false;
 
+  private surveyIdSignal = this.navigationService.getSurveyId();
```

Comment:
The goal was the `navigationService` no longer exposes individual signals for each ID. The calls needs to extract them from UrlParams themselves to ensure they are not accidentally using independent signals to respond to a single event (i.e., the URL changing).